data/transactions/transaction.go

```diff
@@ -660,14 +660,25 @@ type SignedTxGroup struct {
 	// can continue scanning the list from the place where it last stopped.
 	// GroupCounter is local, assigned when the group is first seen by the local transaction pool.
 	GroupCounter uint64
-	// FirstTransactionID is the transaction ID of the first transaction in this transaction group.
-	// TODO - make this more secure by making this the hash of the first signed transaction.
-	FirstTransactionID Txid
+	// GroupTransactionID is the hash of the entire transaction group.
+	GroupTransactionID Txid
 	// EncodedLength is the length, in bytes, of the messagepack encoding of all the transaction
 	// within this transaction group.
 	EncodedLength int
 }
 
+// SignedTxnSlice is a slice of SignedTxn(s), allowing us to
+// easily define the ID() function.
+//msgp:allocbound SignedTxnSlice config.MaxTxGroupSize
+type SignedTxnSlice []SignedTxn
+
+// ID calculate the hash of the signed transaction group.
+func (s SignedTxnSlice) ID() Txid {
```

Comment:
Is this supposed to be the same way we use to find the txgroup hash? If not, should it be?

Reply:
no, it's *supposed* not to be. You see, when finding the regular group hash, we want to calculate the hash without all the signatures, arguments, etc. However, here, we're adding this in order to exchange the transaction groups *before* validating them. That means that we must make sure that no one can provide us with false representation of a transaction group. Hashing it provides that guarantee.

Reply:
If you can think of a better naming for that.. that might be a good thing to do.

Reply:
Okay, this makes sense then. I'm wondering, is there some way to convey that the txns in the slice should correspond to the same group? Or is that not necessary?

Reply:
Interesting idea, but I don't think that the two are strongly correlated.

Reply:
I see, I had this idea since in the comment it says ```// ID calculate the hash of the signed transaction group.```